Build system changes and debian packaging

krnowak opened the pull request:
Filing the PR on behalf of @LiliC (I didn't asked her about that actually, but just wanted to move things forward).

https://phabricator.endlessm.com/T14285

Commit:
move extensions to common dir

Commit:
coverflow: add empty css file

Commit:
gradient: add empty schema

Commit:
coverflow: add Makefile

Commit:
add build files

Commit:
fix schema build

Commit:
glassygnome: add Makefile

Commit:
time_tracker_jsnjack: add Makefile

Commit:
todolist: add Makefile

Commit:
gradientbottombar: add Makefile

Commit:
adjust configure to generate Makefiles

Commit:
coverflow: remove trailing space

Commit:
install all extensions

Commit:
cleanup settings.mk file

Commit:
timetracker: rename EXTENSION_ID

Commit:
glassygnome: rename EXTENSION_ID

Commit:
remove custom installs

Commit:
coverflow: introduce SCHEMA_ID

Commit:
glassygnome: introduce SCHEMA_ID

Commit:
gradient: introduce SCHEMA_ID

Commit:
timetracker: introduce SCHEMA_ID

Commit:
todolist: introduce SCHEMA_ID

Commit:
change dest dir

Commit:
rename package

Commit:
cleanup

Commit:
initial debian

Commit:
todo: compile gschemas

Commit:
remove compile

Commit:
prevent debian from removing all schema files

Commit:
fix package name

Commit:
configure: remove SHELL_VERSION

Commit:
change changelog commit

Commit:
update control

Add github and endless URL. Add short description.

Commit:
gradient: remove empty schema

Commit:
move ENABLED_EXTENSIONS out of configure

Commit:
remove double quotes

Commit:
todolist: modify extension schema lookup

this prevents the need to include precompiled schemas in the extension

Commit:
remove gschema compiling

Commit:
cleanup configure

Commit:
remove unnecessary include

Commit:
add .gitignore to the project

Commit:
add extensions licenses to debian/copyright file

krnowak committed:
update .gitignore

krnowak committed:
Merge pull request #1 from kinvolk/lili/clean_deb

WIP: package up coding shell extensions

iaguis commented:
The changes look good (haven't tested they work but I trust you did).

Maybe we can do some squashing, or at least remove the merge commit which doesn't refer to this repo.

iaguis committed:
Copyright: add missing copyright entries

krnowak committed:
some fixes

add one missing file to the build system, clear some whitespace and
remove unnecessary executable bits from one js file.

erikos commented:
So I had a skim though the commits and they look good. Now with the addition of the License and adding the missing platform.js file we are good to go.

I installed the extensions on my image and run through them and they all work. I played a bit with the settings as well and did toggling with the settings tool that recently landed.

I would say, let's merge and see if the debian packaging works as well...

iaguis reviewed (approved):
Let's do this
